File: facade/c-sharp/CreditCardFacade.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Facade
{
    class CreditCardFacade
    {
        public void CreateCreditCart(Customer customer)
        {
            CreditCardManager creditCardManager = new CreditCardManager(); ;
            BlackListService blackListService = new BlackListService();

            if (!blackListService.CheckEmployeeIsTheBlackList(customer))
            {
                creditCardManager.CreateCreditCart(customer);
            }
        }
    }
}

File: facade/typescript/credit-card-facade.ts
import {Customer} from "./customer";
import {CreditCardManager} from "./credit-card-manager";
import {BlackListService} from "./black-list-service";

export class CreditCardFacade {
  public createCreditCard(customer: Customer): void {
    const creditCardManager = new CreditCardManager();
    const blackListService = new BlackListService();

    if(!blackListService.checkEmployeeIsTheBlackList(customer)) {
      creditCardManager.createCreditCart(customer);
    }
  }
}
